lib/ansible/module_utils/urls.py

```diff
@@ -705,12 +729,33 @@ def _configure_auth(url, url_username, url_password, use_gssapi, force_basic_aut
     return url, headers, handlers
 
 
+_FollowRedirects: _t.TypeAlias = _t.Literal['urllib2', 'all', 'yes', 'safe', 'none']
+
+
 class Request:
-    def __init__(self, headers=None, use_proxy=True, force=False, timeout=10, validate_certs=True,
-                 url_username=None, url_password=None, http_agent=None, force_basic_auth=False,
-                 follow_redirects='urllib2', client_cert=None, client_key=None, cookies=None, unix_socket=None,
-                 ca_path=None, unredirected_headers=None, decompress=True, ciphers=None, use_netrc=True,
-                 context=None):
+    def __init__(
+        self,
+        headers: dict[str, str] | None = None,
+        use_proxy: bool | None = True,
+        force: bool | None = False,
+        timeout: int | None = 10,
+        validate_certs: bool | None = True,
+        url_username: str | None = None,
+        url_password: str | None = None,
+        http_agent: str | None = None,
+        force_basic_auth: bool | None = False,
+        follow_redirects: _FollowRedirects | None = 'urllib2',
+        client_cert: str | None = None,
+        client_key: str | None = None,
+        cookies: cookiejar.CookieJar | None = None,
+        unix_socket: str | None = None,
+        ca_path: str | None = None,
+        unredirected_headers: list[str] | None = None,
+        decompress: bool | None = True,
+        ciphers: list[str] | None = None,
+        use_netrc: bool | None = True,
```

Comment:
Usually, `bool` values mustn't have a union w/ `None` (except for when it's used as a sentinel or has defined semantics).
```suggestion
        use_proxy: bool = True,
        force: bool = False,
        timeout: int | None = 10,
        validate_certs: bool = True,
        url_username: str | None = None,
        url_password: str | None = None,
        http_agent: str | None = None,
        force_basic_auth: bool = False,
        follow_redirects: _FollowRedirects | None = 'urllib2',
        client_cert: str | None = None,
        client_key: str | None = None,
        cookies: cookiejar.CookieJar | None = None,
        unix_socket: str | None = None,
        ca_path: str | None = None,
        unredirected_headers: list[str] | None = None,
        decompress: bool = True,
        ciphers: list[str] | None = None,
        use_netrc: bool = True,
```